packages/deployer/internal/router-ast-validator.js

```diff
@@ -0,0 +1,103 @@
+const { findYulCaseValues, findFunctionSelectors } = require('@synthetixio/core-js/utils/ast');
+const { getModulesSelectors } = require('./contract-helper');
+const { toPrivateConstantCase } = require('./router-helper');
+const filterValues = require('filter-values');
+
+class RouterASTValidator {
+  constructor(asts) {
+    this.asts = asts;
+  }
+
+  async findMissingModuleSelectors() {
+    const moduleSelectors = await getModulesSelectors();
+    const routerSelectors = findYulCaseValues('Router', this.asts['Router']);
+
+    const errors = [];
+    moduleSelectors.forEach((contractSelector) => {
```

Comment:
This can fail with the current implementation of findYulCaseValues (that returns the array or null)

Reply:
No longer a problem cause we return arrays always now